runtime/drivers/clickhouse/transporter_s3.go

```diff
@@ -0,0 +1,58 @@
+package clickhouse
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+	"go.uber.org/zap"
+)
+
+type s3transporter struct {
+	from   drivers.Handle
+	to     drivers.OLAPStore
+	logger *zap.Logger
+}
+
+var _ drivers.Transporter = &s3transporter{}
+
+type sourceProperties struct {
+	URI string `mapstructure:"uri"`
+}
+
+func NewS3Transporter(from drivers.Handle, olap drivers.OLAPStore, logger *zap.Logger) drivers.Transporter {
+	return &s3transporter{
+		from:   from,
+		to:     olap,
+		logger: logger,
+	}
+}
+
+func (t *s3transporter) Transfer(ctx context.Context, srcProps, sinkProps map[string]any, opts *drivers.TransferOptions) error {
+	sinkCfg, err := parseSinkProperties(sinkProps)
+	if err != nil {
+		return err
+	}
+
+	t.logger = t.logger.With(zap.String("source", sinkCfg.Table))
+
+	conf := &sourceProperties{}
+	if err := mapstructure.WeakDecode(srcProps, conf); err != nil {
+		return err
+	}
+
+	// credentials are expected to be added to clickhouse server
+	tableName := fmt.Sprintf("s3_engine_%s_table", safeSQLName(sinkCfg.Table))
+	if err := t.to.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("CREATE TABLE %s ENGINE=S3('%s')", safeSQLName(tableName), conf.URI)}); err != nil {
+		return fmt.Errorf("failed to create table %q with http engine: %w", tableName, err)
+	}
```

Comment:
Yes at this point we are not supporting credentials configured in Rill(Need to decide on the approaches for credentials).
Since clickhouse is running externally we can't take certain decisions like whether to set `use_environment_credentials` or not. Also some parameters like `region` can only be configured directly in server. 
For local, in case clickhouse is running locally nothing changes from user's POV(in comparison to duckDB), the current approach can leverage credentials set in user's local machine. In case clickhouse is running externally, we don't have documented ways to pass credentials to Rill.
For cloud we need to decide on the approach based on how clickhouse will be hosted.

Reply:
This is okay for now, but we should think about how we can set credentials correctly for cases like this in the future (as we look into improving ETL, e.g. we might also want to support transforming GCS data using BigQuery).

In cases where explicit access keys have been set for S3, I think we would be able to cast `t.from` to an S3 driver handle and get the credentials from it? (I think it would be fine not to propagate environment credentials, they are too hard to reason about.)

Reply:
> In cases where explicit access keys have been set for S3, I think we would be able to cast t.from to an S3 driver handle and get the credentials from it?

Yes that should be possible. But we can take it up in next phase.